Sources/Common/Background Queue/QueueRunRequest.swift

```diff
@@ -135,7 +135,14 @@ public class CioQueueRunRequest: QueueRunRequest {
                         self.logger.info("queue is quitting early because all HTTP requests are paused.")
 
                         doneRunning()
-                        break
+                    } else if case .badRequest = error {
+                        self.logger.debug("queue task \(nextTaskStorageId) failed with 400")
+
+                        self.logger.debug("queue deleting task \(nextTaskStorageId) because it will always fail")
+                        _ = self.storage.delete(storageId: nextTaskToRunInventoryItem.taskPersistedId)
+
+                        // since failed task isn't being saved, no need to update `lastFailedTask`
+                        updateWhileLoopLogicVariables(didTaskFail: false, taskJustExecuted: nextTaskToRunInventoryItem)
```

Comment:
🤔 what would happen in this scenario? 

BQ has these tasks in it:
* identify profile
* track event for that profile

identify profile returns a 400. The SDK deletes that task from the BQ and says that it was successful. So, the SDK now attempts to track an event for that profile (that didnt get created on the API) and sends an HTTP request. 

Maybe queue groups need to be accounted for in this work? If a 400 is encountered by a task at the start of a queue group, then all tasks in that group also get deleted?

Reply:
Very good point, thanks for bringing this up.